Source/Rows/StepperRow.swift

```diff
@@ -8,88 +8,154 @@
-        stepper.value = row.value ?? 0
-        stepper.alpha = row.isDisabled ? 0.3 : 1.0
-        valueLabel?.textColor = tintColor
-        valueLabel?.alpha = row.isDisabled ? 0.3 : 1.0
-        valueLabel?.text = row.displayValueFor?(row.value)
-        detailTextLabel?.text = nil
-    }
-
-    @objc func valueChanged() {
-        row.value = stepper.value
-        row.updateCell()
+    selectionStyle = .none
+    stepper.minimumValue = 0
+    stepper.maximumValue = 10
+    stepper.addTarget(self, action: #selector(StepperCell.valueChanged), for: .valueChanged)
+  }
+  
+  open override func update() {
+    super.update()
+    titleLabel.text = row.title
+    titleLabel.isHidden = !shouldShowTitle
+    valueLabel.text = row.displayValueFor?(row.value)
+    stepper.value = row.value ?? 0
+    stepper.isEnabled = !row.isDisabled
+    valueLabel?.textColor = tintColor
+    valueLabel?.alpha = row.isDisabled ? 0.3 : 1.0
+    stepper.alpha = row.isDisabled ? 0.3 : 1.0
+  }
+  
+  @objc func valueChanged() {
+    row.value = stepper.value
+    row.updateCell()
+  }
+  
+  var shouldShowTitle: Bool {
+    return row?.title?.isEmpty == false
+  }
+  
+  private var stepperRow: StepperRow {
+    return row as! StepperRow
+  }
+  
+  open override func updateConstraints() {
+    customConstraints()
+    super.updateConstraints()
+  }
+  
+  open var dynamicConstraints = [NSLayoutConstraint]()
+  
+  open func customConstraints() {
+    guard !awakeFromNibCalled else { return }
+    contentView.removeConstraints(dynamicConstraints)
+    dynamicConstraints = []
+    
+    var views: [String : Any] = ["titleLabel": titleLabel!, "stepper": stepper!, "valueLabel": valueLabel!]
+    let metrics = ["spacing": 15.0]
+    valueLabel.setContentCompressionResistancePriority(.defaultHigh, for: .horizontal)
+    titleLabel.setContentCompressionResistancePriority(.defaultHigh, for: .horizontal)
+    
+    let title = shouldShowTitle ? "[titleLabel]-(>=15@250)-" : ""
+    
+    if let imageView = imageView, let _ = imageView.image {
+      views["imageView"] = imageView
+      let hContraints = NSLayoutConstraint.constraints(withVisualFormat: "H:|-[imageView]-(15)-\(title)[valueLabel]-[stepper]-|", options: .alignAllCenterY, metrics: metrics, views: views)
+      imageView.translatesAutoresizingMaskIntoConstraints = false
+      dynamicConstraints.append(contentsOf: hContraints)
+    } else {
+      let hContraints = NSLayoutConstraint.constraints(withVisualFormat: "H:|-\(title)[valueLabel]-[stepper]-|", options: .alignAllCenterY, metrics: metrics, views: views)
+      dynamicConstraints.append(contentsOf: hContraints)
     }
+    let vContraint = NSLayoutConstraint(item: stepper!, attribute: .centerY, relatedBy: .equal, toItem: contentView, attribute: .centerY, multiplier: 1, constant: 0)
+    dynamicConstraints.append(vContraint)
+    contentView.addConstraints(dynamicConstraints)
+  }
+  
 }
 
-// MARK: StepperRow
 
-open class _StepperRow: Row<StepperCell> {
```

Comment:
You should leave an open superclass to allow overrides

Reply:
sure, that comes from SliderView - ok if I fix it there too then?

Reply:
Yes